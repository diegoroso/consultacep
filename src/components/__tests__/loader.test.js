import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Loader from '../loader'

Enzyme.configure({ adapter: new Adapter() })

const setup = () => {
  const props = {
    show: false
  }

  const enzymeWrapper = shallow(<Loader {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('Loader', () => {
  const { enzymeWrapper } = setup()
  expect(enzymeWrapper).toMatchSnapshot();

  it('Aparecer e esconder o loader quando necessÃ¡rio', () => {
    expect(enzymeWrapper.find('.loader').hasClass('show')).toBeFalsy()
    enzymeWrapper.setProps({ show: true })
    expect(enzymeWrapper.find('.loader').hasClass('show')).toBeTruthy()
  })
})
