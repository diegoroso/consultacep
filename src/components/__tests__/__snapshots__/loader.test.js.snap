// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader
    show={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="loader__content"
      >
        <img
          alt="Carregando"
          src="file-mock-for-components-test"
        />
        <p>
          Procurando localização no mapa
        </p>
      </div>,
      "className": "loader ",
      "data-cy": "Loader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Carregando"
            src="file-mock-for-components-test"
          />,
          <p>
            Procurando localização no mapa
          </p>,
        ],
        "className": "loader__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Carregando",
            "src": "file-mock-for-components-test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Procurando localização no mapa",
          },
          "ref": null,
          "rendered": "Procurando localização no mapa",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loader__content"
        >
          <img
            alt="Carregando"
            src="file-mock-for-components-test"
          />
          <p>
            Procurando localização no mapa
          </p>
        </div>,
        "className": "loader ",
        "data-cy": "Loader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Carregando"
              src="file-mock-for-components-test"
            />,
            <p>
              Procurando localização no mapa
            </p>,
          ],
          "className": "loader__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Carregando",
              "src": "file-mock-for-components-test",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Procurando localização no mapa",
            },
            "ref": null,
            "rendered": "Procurando localização no mapa",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
